needy/platform.py

```diff
@@ -45,11 +43,17 @@ def configuration_hash(self, architecture):
             pass
 
         hash = hashlib.sha256()
-        hash.update(json.dumps({
+        hash_dict = {
             'c-compiler': cc,
             'cxx-compiler': cxx,
-        }, sort_keys=True).encode())
-        return hash.digest()
+        }
+        hash.update(json.dumps(hash_dict, sort_keys=True).encode())
+        digest = hash.digest()
+        logging.debug('configuration hash {} formed from...\n{}'.format(
+            binascii.hexlify(digest),
+            json.dumps(hash_dict, sort_keys=True, indent=4, separators=(',', ': ')))
+        )
```

Comment:
i think there are a few code paths where this gets invoked twice. i'd rather not see this in the logs twice though or have to think in the future about where and how this is called.

is there somewhere more closely related to your use-case where you can put this? maybe it'd make sense to be in the debug output for the status command?